.github/workflows/storage.yml

```diff
@@ -156,3 +156,37 @@ jobs:
       run: scripts/setup_bundler.sh
     - name: PodLibLint Storage Cron
       run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseStorage.podspec --platforms=${{ matrix.target }} --use-static-frameworks --skip-tests
+
+  doc-diff:
+    # Run on PRs only.
+    if: github.event_name == 'pull_request'
+    runs-on: macos-12
+    steps:
+    - name: Create artifacts folder
+      run: mkdir artifacts && echo "ARTIFACTS_DIR=$(pwd)/artifacts" >> $GITHUB_ENV
+    - name: Fetch all
+      run: git clone https://github.com/firebase/firebase-ios-sdk.git
+    - uses: ruby/setup-ruby@master
+      with:
+        working-directory: firebase-ios-sdk
+    - name: Setup Bundler
+      working-directory: firebase-ios-sdk
+      run: scripts/setup_bundler.sh
+    - name: Generate workspace (main)
+      working-directory: firebase-ios-sdk
+      run: git checkout master && bundle exec pod gen FirebaseStorage.podspec --local-sources=./ --platforms=ios
+    - name: Build documentation
+      working-directory: firebase-ios-sdk
+      run: xcodebuild docbuild -workspace gen/FirebaseStorage/FirebaseStorage.xcworkspace -scheme FirebaseStorage -sdk iphonesimulator -derivedDataPath ${{ env.ARTIFACTS_DIR }}/main/
```

Comment:
Since iOS docs are almost always a superset, should we deal with other platforms as notes in the actual doc source?